### PR TITLE
Parameter box improvements

### PR DESCRIPTION
**Pull request by yw4z**
### Adds missing / empty side texts
• Added "Layers" side text for
> Top shell layers
> Bottom shell layers
> Number of slow layers

• Added "Layer" side text for
> Full fan speed at layer

• Added "mm" side text for
> Mesh margin
> Minimum wall length

• Added "°C" side text for
> Softening temperature

### Fixes combo boxes without arrows
Support > Style
Screenshot-20240423155322

Filament Supports
Screenshot-20240423155447

Support > Advanced
Screenshot-20240423155536
Screenshot-20240423155541

after fix shows dropdown arrows if there is no icon for dropdown item

Screenshot-20240427173425

### Converts double input boxes to regular input boxes

Screenshot-20240424134934

Screenshot-20240424134941

Screenshot-20240424134956

After Fix
Screenshot-20240524184001
• Used icons for X / Y instead text. input boxes aligns perfectly with other components
• Didn't use combined side text because it not matches style with other components
• Solves styling problem and gives same result on all platforms

**Preview for Printer Settings > Printable Area > Set**

| before | after |
| ------------- | ------------- |
| Screenshot-20240427173214 | Screenshot-20240427172935 |

**Also matched style of Create printer window**
• Used text align on left like on other components
• Matched width of input boxes with other components

| before | after |
| ------------- | ------------- |
| Screenshot-20240427173123 | Screenshot-20240427173049|

### Questionable areas for discussion (EXCLUDED FROM COMMIT)

• Converting these to spin boxes. Currently it combines combo box and input box and it has weird usage and no need to use on this ones because dropdown only shows numbers without any description. Using spin boxes better because other layer related input boxes uses this too
> Top interface layers
> Bottom interface layers

Screenshot-20240423021048

Screenshot-20240423021146

I noticed there is a "Same as Top" option for Bottom interface layers. I guess i will revert back to default with your comment

not rendered well with side text
Screenshot-20240423181955

solution with spin box. there is an explanation required on tooltip for -1 equals same as top
Screenshot-20240423182156

Also this ones combines combo box and input box. like i said its weird but helpful on this ones compared to interface layers ones
> sparse_infill_line_width
> infill_anchor_max
> infill_anchor

If you planning to add more pre defined values for value boxes in future maybe increasing width of input boxes will be a good choice because value + dropdown arrow + side text looks very dense

Screenshot-20240423014410

I tried to make use them as regular input box in my UI commit but probably revert back with your comment

### CODE REVIEW

**Comment by discip** [comment]
1. > • Added "Layers" side text for

   > • Added "Layer" side text for

   Regarding all the _'side texts'_, you should consider the context:

   If the corresponding unit is used in the description <ins>in front</ins> of the value, there is no need to put it **additionally** thereafter.

   image
   ##
2. > • Added "mm" side text for

   &

   > • Added "°C" side text for

   ## 👍

3. > • Converted these to spin boxes ...

   * Due to the mentioned reason in **`1.`**, no need for _'side texts'_ here as well.
   * Since a drop-down allows you to access all the values ​​more quickly without having to click through them individually, as would be the case using a spin box, please leave this as is.
   
4. > I tried to make use them as regular input box in my UI commit but probably revert back with your comment

   Please don't remove the arrow.

**Comment by yw4z** [comment]
@discip then we should delete this with your logic

Screenshot-20240423135607

side text generally uses units and its unit "Layer" / "Layers" in my perspective

problem is combining combo box and and input box and parameter value boxes too narrow to fit multiple components

**Comment by discip** [comment]
1. > then we should delete these with your logic

   Absolutely! 😃👍
   I haven't checked every other possible case, but since you're at it, it would be great if you maintained the tendency you've demonstrated to make everything as consistent as possible. 😊

2. > side text generally uses units and its unit "Layer" / "Layers" in my perspective

   To demonstrate what I mean:
   Compared to e.g.: **`length`**, where you could have **`mm`** or **`cm`**, in which case it makes perfect sense to put the unit in the text box, even though the description contains the word length, the _'unit'_ **`layers`** doesn't need any further specification.

3. > problem is combining combo box and and input box and parameter value boxes too narrow to fit multiple components

   I'm having no issues using the current setup. 🤷🏻

4. > about spin boxes you can set value faster with them compared to combo box

   1. It depends on the amount of values the list contains.
   2. In most cases you would need 2 clicks to get _**any value**_ with a **`drop down`** opposed to a **`spin box`**.
   e.g.: If your current value is **`1`** you would need to click 4 times to get to the **`5`** with a **`spin box`**. With a **`drop down`** however, you click the arrow and **`5`** (2 clicks)!

I'm really curious about what others think about these changes.

**Comment by yw4z** [comment]
> 4. > about spin boxes you can set value faster with them compared to combo box
>    1. It depends on the amount of values the list contains.
>    2. In most cases you would need 2 clicks to get _**any value**_ with a **`drop down`** opposed to a **`spin box`**.
>       e.g.: If your current value is **`1`** you would need to click 4 times to get to the **`5`** with a **`spin box`**. With a **`drop down`** however, you click the arrow and **`5`** (2 clicks)!

you just count mouse clicks. you should also count mouse movements as steps

**Comment by yw4z** [comment]
@SoftFever this commit also ready for review. also there is a "Questionable areas for discussion" section on first post

**Comment by yw4z** [comment]
@SoftFever should i split this commit to separate commits
• Add missing side texts for parameter boxes 
• Fix combo box arrows not appearing if there is no icon for drop down list item
• Convert double input boxes for points to regular input boxes
• Better color for disabled elements on dark mode (Will move this to new color fixes commit)
• Match text color of disabled combo box (Will move this to new color fixes commit)

File: src/libslic3r/PrintConfig.cpp [hunk 1]
```diff
@@ -816,6 +816,7 @@ void PrintConfigDef::init_fff_params()
     def = this->add("bottom_shell_layers", coInt);
     def->label = L("Bottom shell layers");
     def->category = L("Strength");
+    def->sidetext = L("layers"); // ORCA add side text
```
Comment (SoftFever) [inline src/libslic3r/PrintConfig.cpp:819]: This works really good 👍

File: src/libslic3r/PrintConfig.cpp [hunk 2]
```diff
@@ -918,6 +919,7 @@ void PrintConfigDef::init_fff_params()
     def->category = L("Quality");
     def->tooltip = L("Decrease this value slightly(for example 0.9) to reduce the amount of material for bridge, "
                      "to improve sag");
+    def->sidetext = L("x"); // ORCA add side text
```
Comment (SoftFever) [inline src/libslic3r/PrintConfig.cpp:922]: I find add "x" side text strange.
Can we remove x side texts?

File: resources/images/inputbox_y.svg
Comment (SoftFever) [inline resources/images/inputbox_y.svg:1]: Good one!
One suggestion though, can we make the font easier to read? the current square font may cause confusion IMO

File: src/libslic3r/PrintConfig.cpp (hunk 2 above)
Comment (yw4z) [reply src/libslic3r/PrintConfig.cpp:922]: there is a % used for percentage. x fits well for ratio imo. also i tried to show values as "1.00" to indicate its float numbers but they always rounded to "1"

**Comment by yw4z** [comment]
@SoftFever updated point control icons
before
Screenshot-20240524182305

after
Screenshot-20240524184001

File: src/libslic3r/PrintConfig.cpp (hunk 2 above)
Comment (SoftFever) [reply src/libslic3r/PrintConfig.cpp:922]: `%` and other units like `mm` are common and therefore look natural.
The x looks very strange and confusing as it displays like `1     x`.
We shouldn't add it.

Comment (yw4z) [reply src/libslic3r/PrintConfig.cpp:922]: google suggest few alternatives for multiplier symbol.
Multiplication sign definition: the symbol ( ⋅ ), ( × ), or ( ∗ )
But ratio is completely different thing. it mostly suggest proportional ratio and its symbol is " : "

```
1     mm
1     x
1     ⋅
1     ×
1     ∗
```

ofc i would not die if we dont add it. still looks nice imo :)

here is degree, mm, % and x in same view

Screenshot-20240525033427

Comment (SoftFever) [reply src/libslic3r/PrintConfig.cpp:922]: I still think the x is confusing here :)

**Comment by SoftFever** [comment]
@yw4z 
What's the status of the PR?
Could we remove the `x` sidenote?

**Comment by yw4z** [comment]
@SoftFever yes i will remove them but quite busy atm. probably complete changes on PRs on end of week

**Comment by yw4z** [comment]
@SoftFever reverted changes for "x" side text and removed color related fixes because they need additional fixes. i will send color related issues as another commit because i found 10+ more

**Review by SoftFever — APPROVED** [review]
LGTM

**Pull request merged**